api/types/role.go

```diff
@@ -144,44 +146,7 @@ const (
 // Equals returns true if the roles are equal. Roles are equal if options,
 // namespaces, logins, labels, and conditions match.
 func (r *RoleV3) Equals(other Role) bool {
-	if !r.GetOptions().Equals(other.GetOptions()) {
-		return false
-	}
-
-	for _, condition := range []RoleConditionType{Allow, Deny} {
-		if !utils.StringSlicesEqual(r.GetLogins(condition), other.GetLogins(condition)) {
-			return false
-		}
-		if !utils.StringSlicesEqual(r.GetNamespaces(condition), other.GetNamespaces(condition)) {
-			return false
-		}
-		if !r.GetNodeLabels(condition).Equals(other.GetNodeLabels(condition)) {
-			return false
-		}
-		if !r.GetAppLabels(condition).Equals(other.GetAppLabels(condition)) {
-			return false
-		}
-		if !r.GetDatabaseLabels(condition).Equals(other.GetDatabaseLabels(condition)) {
-			return false
-		}
-		if !utils.StringSlicesEqual(r.GetDatabaseNames(condition), other.GetDatabaseNames(condition)) {
-			return false
-		}
-		if !utils.StringSlicesEqual(r.GetDatabaseUsers(condition), other.GetDatabaseUsers(condition)) {
-			return false
-		}
-		if !RuleSlicesEqual(r.GetRules(condition), other.GetRules(condition)) {
-			return false
-		}
-		if !r.GetClusterLabels(condition).Equals(other.GetClusterLabels(condition)) {
-			return false
-		}
-		if !r.GetKubernetesLabels(condition).Equals(other.GetKubernetesLabels(condition)) {
-			return false
-		}
-	}
-
-	return true
+	return cmp.Equal(r, other, cmpopts.IgnoreFields(RoleV3{}, "Kind", "SubKind", "Version", "Metadata"))
```

Comment:
I can understand why `Metadata` and perhaps `Version` need to be excluded but why `Kind` and `SubKind`?

Also given that this equal condition is different from the other resources whose `Equal` functions are generated by the proto plugin, wouldn't it make sense to name it differently so that every call site immediately indicates it's not a full all-field comparison?

Reply:
Turns out this method wasn't needed at all.